diff --git a/testbench/tb_top.sv b/testbench/tb_top.sv
index 7f5f600..6125094 100644
--- a/testbench/tb_top.sv
+++ b/testbench/tb_top.sv
@@ -15,12 +15,8 @@
 //
 // this is testbench file
 
-`ifdef VERILATOR
-module tb_top ( input bit core_clk );
-`else
 module tb_top;
     bit                         core_clk;
-`endif
     logic                       rst_l;
     logic                       porst_l;
     logic                       nmi_int;
@@ -433,8 +429,8 @@ module tb_top;
 
 `ifndef VERILATOR
         if($test$plusargs("dumpon")) $dumpvars;
-        forever  core_clk = #5 ~core_clk;
 `endif
+        forever  core_clk = #5 ~core_clk;
     end
 
 
diff --git a/testbench/test_tb_top.cpp b/testbench/test_tb_top.cpp
index 899caf1..6908614 100644
--- a/testbench/test_tb_top.cpp
+++ b/testbench/test_tb_top.cpp
@@ -22,7 +22,7 @@
 #include "verilated_vcd_c.h"
 
 
-vluint64_t main_time = 0;
+double main_time = 0;
 
 double sc_time_stamp () {
  return main_time;
@@ -50,8 +50,7 @@ int main(int argc, char** argv) {
 #if VM_TRACE
       tfp->dump (main_time);
 #endif
-      main_time += 5;
-      tb->core_clk = !tb->core_clk;
+      main_time = tb->nextTimeSlot();
       tb->eval();
   }
 
